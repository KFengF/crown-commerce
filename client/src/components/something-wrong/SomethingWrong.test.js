import React from 'react';
import { shallow } from 'enzyme';
import SomethingWrong from './SomethingWrong.component';
import SomethingWrongHandler from './SomethingWrong.handler';

describe('SomethingWrong component', () => {
    it('should render SomethingWrong component', () => {
        expect(shallow(<SomethingWrong />)).toMatchSnapshot();
    });
});

describe('SomethingWrong handler', () => {
    let TestComponent;
    let error;
    let wrapper;
    const originalError = console.error;
    //Usar esto para evitar los errores

    beforeEach(() => {
        console.error = () => {};
        TestComponent = () => <div id="test" />;
        error = new Error('An intended error');
        wrapper = shallow(
            <SomethingWrongHandler>
                <TestComponent />
            </SomethingWrongHandler>
        );
    });

    afterAll(() => {
        console.error = originalError;
    });
    
    describe('if there is not an error', () => {
        it('should render TestComponent', () => {
            expect(wrapper.exists(TestComponent)).toBe(true);
        });

        it('should not render SomethingWrong handler', () => {
            expect(wrapper.exists(SomethingWrong)).toBe(false);
        });
    });

    describe('if there is an error', () => {
        it('should render SomethingWrong handler', () => {
            wrapper.find(TestComponent).simulateError(error);
            expect(wrapper.exists(SomethingWrong)).toBe(true);
        });

        it('should not render TestComponent', () => {
            wrapper.find(TestComponent).simulateError(error);
            expect(wrapper.exists(TestComponent)).toBe(false);
        });
    });
});