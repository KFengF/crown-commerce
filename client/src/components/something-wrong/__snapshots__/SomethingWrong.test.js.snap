// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SomethingWrong component should render SomethingWrong component 1`] = `ShallowWrapper {}`;
